### PR TITLE
@damassi => Add component for pageviews trust signal for an artwork

### PR DESCRIPTION
**Pull request by mzikherman**
Still going to add a spec here.

Looks like:

screen shot 2018-11-06 at 2 13 46 pm

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit 4c40d71545dd11d1e022ab00bd90ee37e4f1a760

https://deploy-preview-1499--artsy-reaction.netlify.com

File: src/Apps/Artwork/Components/Pageviews.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import { renderWithLoadProgress } from "Artsy/Relay/renderWithLoadProgress"
+import React, { Component } from "react"
+import { createFragmentContainer, graphql, QueryRenderer } from "react-relay"
+import { PageviewsIcon } from "Styleguide/Elements/icons/Pageviews"
+
+interface Props {
+  artwork: Pageviews_artwork
+}
+
+export class Pageviews extends Component<Props> {
```
Comment (damassi) [inline src/Apps/Artwork/Components/Pageviews.tsx]: SFC

File: src/Apps/Artwork/Components/Pageviews.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    if (pageviews) {
+      return (
+        <>
+          <PageviewsIcon width="21" height="21" />
+          <div
+            style={{
+              display: "inline-block",
+              verticalAlign: "sub",
+              marginLeft: "15px",
+            }}
```
Comment (damassi) [inline src/Apps/Artwork/Components/Pageviews.tsx]: Please replace with flex or box and avoid style tags completely

File: src/Apps/Artwork/Components/Pageviews.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+          >
+            <Sans weight="medium" size="2">
+              This work is getting noticed!
+            </Sans>
+            <Sans color={color("black60")} size="2" weight="regular">
+              It has been viewed{" "}
+              <span style={{ fontWeight: "bold" }}>
+                {pageviews.toLocaleString()}
+              </span>{" "}
```
Comment (damassi) [inline src/Apps/Artwork/Components/Pageviews.tsx]: no span, use our fonts...

File: src/Apps/Artwork/Components/Pageviews.tsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    } else {
+      return (
+        <>
+          <PageviewsIcon width="21" height="21" />
+          <div
+            style={{
+              display: "inline-block",
+              verticalAlign: "sub",
+              marginLeft: "15px",
+            }}
```
Comment (damassi) [inline src/Apps/Artwork/Components/Pageviews.tsx]: same

File: src/Apps/Artwork/Components/__stories__/ArtworkSidebar/ArtworkSidebarPageviews.story.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    <React.Fragment>
+      <Section title="Artwork with > 10 views">
+        <Box width="100%">
+          <ArtworkSidebarPageviewsQueryRenderer artworkID="andy-warhol-skull" />
+        </Box>
+      </Section>
+      <Section title="Artwork with < 10 views">
+        <Box width="100%">
+          <ArtworkSidebarPageviews
+            artwork={{ id: "blah", pageviews: null } as any}
```
Comment (damassi) [inline src/Apps/Artwork/Components/__stories__/ArtworkSidebar/ArtworkSidebarPageviews.story.tsx]: does pageviews need to be explicitly `null`?

Comment (mzikherman) [reply src/Apps/Artwork/Components/__stories__/ArtworkSidebar/ArtworkSidebarPageviews.story.tsx]: Yea, that's what we return from MP if we don't want to display the explicit amount and accompanying text, but the other version.

File: src/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarPageviews.tsx [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+export class ArtworkSidebarPageviews extends Component<Props> {
+  render() {
+    const { artwork } = this.props
+    const { pageviews } = artwork
+
+    // TODO: Figure out why `jest` tests don't play nicely with this icon.
+    if (pageviews) {
+      return (
+        <>
+          {typeof jest === "undefined" && (
```
Comment (mzikherman) [inline src/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarPageviews.tsx]: Unfortunately this icon seems to work fine (in the browser), and the SVG looks fine, but doesn't play well with our usual testing, so am leaving out for now.

**Comment by mzikherman** [comment]
Updated with a test.

File: src/Apps/Artwork/Components/Pageviews.tsx (hunk 1 above)
Comment (mzikherman) [reply src/Apps/Artwork/Components/Pageviews.tsx]: good call! (this could maybe be a Danger rule even)

Comment (damassi) [reply src/Apps/Artwork/Components/Pageviews.tsx]: Its more of a linter rule -- if component doesn't contain state, it should be an SFC. (I'm sorry for always flagging this in PR after PR, but basically everything should be an SFC and the upcoming React version effectively deprecates classes.)

File: src/Apps/Artwork/Components/Pageviews.tsx (hunk 3 above)
Comment (mzikherman) [reply src/Apps/Artwork/Components/Pageviews.tsx]: this just bolds a word within a `Sans` tho...I rewrote the other parts to use `Flex` and `Box` (which btw was way easier/nicer than my initial attempt), but I'm not sure how to rewrite this?

Comment (damassi) [reply src/Apps/Artwork/Components/Pageviews.tsx]: `<Sans weight='medium'>{pageviews.toLocaleString()}</Sans>` (or bold if that's available as a weight)

Comment (damassi) [reply src/Apps/Artwork/Components/Pageviews.tsx]: Or even just `<strong>something</strong>`

**Comment by mzikherman** [comment]
Thanks for the nudge to use palette, was incredibly easy with `Flex` and `Box`, the correct spacing to `ml`.

File: src/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarPageviews.tsx [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+
+interface Props {
+  artwork: ArtworkSidebarPageviews_artwork
+}
+
+export const ArtworkSidebarPageviews: SFC<Props> = props => {
+  const { artwork } = props
+  const { pageviews } = artwork
+
+  // TODO: Figure out why `jest` tests don't play nicely with this icon.
```
Comment (damassi) [inline src/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarPageviews.tsx]: 👍

File: src/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarPageviews.tsx [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+        {typeof jest === "undefined" && (
+          <Box mt={0.5}>
+            <PageviewsIcon width="21" height="21" />
+          </Box>
+        )}
+        <Box ml={2}>
+          <Sans weight="medium" size="2">
+            This work is getting noticed!
+          </Sans>
+          <Sans color={color("black60")} size="2" weight="regular">
```
Comment (damassi) [inline src/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarPageviews.tsx]: `<Sans color='black60'>`

File: src/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarPageviews.tsx [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+        {typeof jest === "undefined" && (
+          <Box mt={0.5}>
+            <PageviewsIcon width="21" height="21" />
+          </Box>
+        )}
+        <Box ml={2}>
+          <Sans weight="medium" size="2">
+            You're one of the first to view this work
+          </Sans>
+          <Sans color={color("black60")} size="2" weight="regular">
```
Comment (damassi) [inline src/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarPageviews.tsx]: same here

**Comment by mzikherman** [comment]
I think I addressed everything, thanks for the push!

**Pull request merged**